core/src/main/java/org/apache/iceberg/V2Metadata.java

```diff
@@ -254,7 +254,8 @@ static Schema wrapFileSchema(Types.StructType fileSchema) {
         DataFile.UPPER_BOUNDS,
         DataFile.KEY_METADATA,
         DataFile.SPLIT_OFFSETS,
-        DataFile.EQUALITY_IDS
+        DataFile.EQUALITY_IDS,
+        DataFile.NAN_VALUE_COUNTS
```

Comment:
Similar to above, I think we should add NaN value counts just below null value counts. This will affect the ordinals in `get` so you'll have to update them.